lms/static/js/spec/utils/dropdown_spec.js

```diff
@@ -0,0 +1,122 @@
+                });
+
+                expect(menu).toHaveAttr({
+                    'tabindex': '-1'
+                });
+            });
+        });
+
+        describe('when running', function() {
+            function keyPressEvent(key) {
+                return $.Event('keydown', { keyCode: key });
+            }
+
+            var wrapper, button, menu, menu_item, menu_action, KEY = $.ui.keyCode,
+
+            it('opens the menu on button click', function() {
+                button.click();
+                expect(button).toHaveClass('is-active');
+                expect(button).toHaveAttr({
+                    'aria-expanded': 'true'
+                });
+                expect(menu).toHaveClass('is-visible');
+            });
+
+            it('closes the menu on outside click', function() {
+                $(window).click();
+                expect(button).not.toHaveClass('is-active');
+                expect(button).toHaveAttr({
+                    'aria-expanded': 'false'
+                });
+                expect(menu).not.toHaveClass('is-visible');
+                expect(menu).toHaveClass('is-hidden');
+            });
+
+            it('opens the menu on ENTER kepress', function() {
+                button.trigger(keyPressEvent(KEY.ENTER));
+                expect(button).toHaveClass('is-active');
+                expect(button).toHaveAttr({
+                    'aria-expanded': 'true'
+                });
+                expect(menu).toHaveClass('is-visible');
+                expect(menu.focus).toHaveBeenCalled();
+            });
+
+            it('opens the menu on DOWN keypress', function() {
+                button.trigger(keyPressEvent(KEY.DOWN));
+                expect(button).toHaveClass('is-active');
+                expect(button).toHaveAttr({
+                    'aria-expanded': 'true'
+                });
+                expect(menu).toHaveClass('is-visible');
+                expect(menu.focus).toHaveBeenCalled();
+            });
+
+            it('moves between menu items on UP or DOWN', function() {
+                var last_item = menu_item.length - 1, i;
+                container.trigger(keyPressEvent(KEY.ENTER));
+
+                for (i = last_item; i >= 0; i--) {
+                    menu_item.eq(i).trigger(keyPressEvent(KEY.UP));
+                }
+
+                for (i = 0; i <= last_item; i++) {
+                    menu_item.eq(i).trigger(keyPressEvent(KEY.DOWN));
+                }
+            });
+
+            it('opens the menu on SPACE kepress', function() {
+                button.trigger(keyPressEvent(KEY.SPACE));
+                expect(button).toHaveClass('is-active');
+                expect(button).toHaveAttr({
+                    'aria-expanded': 'true'
+                });
+                expect(menu).toHaveClass('is-visible');
+                expect(menu.focus).toHaveBeenCalled();
+            });
+
+            it('closes the menu on ESC keypress', function() {
```

Comment:
You handle ESC in 2 different places-- handlerIsAction and directly in listenForKeypress. Only one of those 2 paths are tested here. Test the other one as well.

And then in general, make sure that every "if" clause in handlerIsAction, handlerIsButton, and handlerIsMenu is covered by a test.